elyra/cli.py

```diff
@@ -0,0 +1,135 @@
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        response = await PipelineProcessorManager.instance().process(pipeline)
+
+    return response.to_json()
+
+
+def prepare_pipeline(pipeline_path, name=None, runtime='local', runtime_config='local'):
+    pipeline_abs_path = os.path.join(os.getcwd(), pipeline_path)
+    pipeline_dir = os.path.dirname(pipeline_abs_path)
+
+    if name is None:
+        name = os.path.splitext(os.path.basename(pipeline_abs_path))[0]
+
+    with open(pipeline_abs_path) as f:
+        pipeline_definition = json.load(f)
+
+    for pipeline in pipeline_definition["pipelines"]:
+        for node in pipeline["nodes"]:
+            if node["app_data"]["filename"]:
+                abs_path = os.path.join(os.getcwd(), pipeline_dir, node["app_data"]["filename"])
+                node["app_data"]["filename"] = abs_path
+
+    # NOTE: The frontend just set the info for first pipeline, but shouldn't it
+    # search for the primary pipeline and set that?
+    # Setting `pipeline_definition["pipelines"][0]` for consistency.
+    pipeline_definition["pipelines"][0]["app_data"]["name"] = name
+    pipeline_definition["pipelines"][0]["app_data"]["runtime"] = runtime
+    pipeline_definition["pipelines"][0]["app_data"]["runtime-config"] = runtime_config
+
+    return pipeline_definition
+
+
+def print_banner(title):
+    click.echo(Fore.CYAN + "────────────────────────────────────────────────────────────────" + Style.RESET_ALL)
+    click.echo(Fore.CYAN + "  {}".format(title) + Style.RESET_ALL)
+    click.echo(Fore.CYAN + "────────────────────────────────────────────────────────────────" + Style.RESET_ALL)
+    click.echo()
+
+
+def print_info(title, info_list):
+    click.echo(Fore.CYAN + "❯ {}".format(title) + Style.RESET_ALL)
+    for info_item in info_list:
+        if isinstance(info_item, str):
+            click.echo("  {}".format(info_item))
+        else:
+            click.echo("  {}: {}".format(info_item[0], info_item[1]))
+    click.echo()
+
+
+def print_version():
+    print_info("Version", ["elyra {}".format(__version__)])
+
+
+@click.group()
+def cli():
+    pass
+
+
+@click.command()
+@click.argument('pipeline_path', type=click.Path(exists=True))
+@click.option('-n', '--name')
+def submit(pipeline_path, name):
+    """TODO: Description"""
+    click.echo()
+
+    print_banner("Elyra Pipeline Submission")
+    print_version()
+
+    pipeline_definition = prepare_pipeline(pipeline_path, name=name, runtime='kfp', runtime_config='fun')
+    pipeline = PipelineParser().parse(pipeline_definition)
+
+    print_info("Info", [("name", pipeline.name)])
+    print_info("Runtime", [("type", pipeline.runtime), ("config", pipeline.runtime_config)])
+
+    with yaspin(text="Submitting Pipeline..."):
+        os.environ["ELYRA_METADATA_PATH"] = os.path.join(os.path.expanduser("~"), ".elyra")
+        msg = asyncio.get_event_loop().run_until_complete(submit_pipeline(pipeline))
```

Comment:
@kevin-bates I'm very new to python, is this how to properly run an async function? how do I handle errors? `try`/`except` doesnt seem to work right

Reply:
nvm I was listening for the wrong error

Reply:
Hmm - that looks right and I have a small sample I updated to model this who's try/except is properly behaved.  If you wrap the `run_until_complete` in a try/except, what happens on exceptions?

Reply:
Sorry - I hadn't seen your response - glad things are moving forward.